Homework Daniil Bannov

bannovdaniil opened the pull request:
Задание, выполнил.
Тесты проходит.

bannovdaniil committed:
1. реализация счетчика

bannovdaniil committed:
2. прошел все тесты.

bannovdaniil committed:
3. Init jersey server.

bannovdaniil committed:
4. Переделал на JAXRS_APPLICATION_CLASS.

bannovdaniil committed:
5. все jersey тесты

bannovdaniil committed:
6. рефакторинг

bannovdaniil committed:
7. more time for test on gitHub

bannovdaniil committed:
7. restored time

bannovdaniil committed:
8. Atomic to LongAdder

k-vasiliev commented on jersey/src/main/java/resource/CounterResource.java:
```diff
@@ -0,0 +1,66 @@
+  }
+
+  @GET
+  @Path("/status")
```
В отдельный бы ресурс, тогда спокойно над всем ресурсом повесил бы `@Path("/counter")`

k-vasiliev commented on jersey/src/main/java/resource/CounterResource.java:
```diff
@@ -0,0 +1,66 @@
+      }
+    }
+    return Response
+        .status(Response.Status.BAD_REQUEST)
```
тут не очень удобно написано
Если у тебя появятся новые условия, то скорее всего diff будет такой, что затронет весь метод

```
if (StringUtil.isBlank(deltaValue)) {
    return Response
        .status(Response.Status.BAD_REQUEST)
        .entity("Number is absent.")
        .build();
}
```

можно сразу вернуть ошибку и избавиться от лишней вложенности для try-catch

k-vasiliev commented on jersey/src/main/java/service/CounterService.java:
```diff
@@ -0,0 +1,32 @@
+import dto.CounterDto;
+
+public class CounterService {
+  private final static CounterDao counterDao = CounterDao.getInstance();
```
Есть смысл его тут как static хранить?

bannovdaniil replied:
это я при тестировании добавил, когда тесты не проходило, подумал, будет со статикой быстрее. :)
а так нет, он SingleTone, 
хотя както читал статью, что даже синглТоны в многопоточной среде могут до нескольких раз успеть создаться.

k-vasiliev replied:
нужны тредсейф синглтоны :)

bannovdaniil committed:
Merge branch 'master' into develop

bannovdaniil committed:
11. fix: NPE cookie

bannovdaniil committed:
12. refactoring: убрал синхронайз, когда пробовал просто Long

k-vasiliev commented on servlet/src/main/java/resourse/ResourceCounter.java:
```diff
@@ -0,0 +1,102 @@
+            "hh-auth".equals(cookie.getName())
+                && StringUtil.isNotBlank(cookie.getValue())
+                && cookie.getValue().length() > 10
+        ).findFirst()
```
Можно `anyMatch()`

bannovdaniil replied:
верно! нам же не надо значение куки

bannovdaniil committed:
Merge remote-tracking branch 'k-vasiliev/add-junit-reports' into develop

bannovdaniil committed:
Merge remote-tracking branch 'k-vasiliev/add-junit-reports' into develop

bannovdaniil committed:
Merge remote-tracking branch 'k-vasiliev/add-junit-reports' into develop

bannovdaniil committed:
13. работа над ошибками:
1. убрал лишнюю вложенность.
2. убрал static для SingleTone
3. findFirst -> anyMatch

bannovdaniil committed:
Merge remote-tracking branch 'origin/master' into develop

bannovdaniil committed:
Merge remote-tracking branch 'origin/master' into develop

bannovdaniil committed:
14. сломал тест -delta > +delta

bannovdaniil committed:
15. рабочая версия.
